os/arch/arm/src/armv8-m/up_doirq.c

```diff
@@ -101,22 +128,41 @@ uint32_t *up_doirq(int irq, uint32_t *regs)
 	 * that purpose as implemented here because only the outermost nested
 	 * interrupt can result in a context switch (it can probably be deleted).
 	 */
-
-	/* Current regs non-zero indicates that we are processing an interrupt;
-	 * current_regs is also used to manage interrupt level context switches.
-	 */
-
 	savestate = (uint32_t *)current_regs;
 	current_regs = regs;
-
 	/* Acknowledge the interrupt */
 
 	up_ack_irq(irq);
+#endif
 
 	/* Deliver the IRQ */
 
 	irq_dispatch(irq, regs);
 
+#ifdef CONFIG_ARCH_NESTED_INTERRUPT
+	/* Context switches are indicated by the returned value of this function.
+	 * If a context switch occurred while processing the interrupt then
+	 * current_regs may have change value.  If we return any value different
+	 * from the input regs, then the lower level will know that a context
+	 * switch occurred during interrupt processing.  Context switching should
+	 * only be performed when the outermost interrupt handler returns.
+	 */
+
+	flags = irqsave();
+
+	g_nestlevel--;
+
+	if (g_nestlevel == 0) {
+		regs = (uint32_t*)current_regs;
```

Comment:
Please add space between `uint32_t` and `*`.

Reply:
fixed in 19305da